ir/state.cpp

```diff
@@ -199,15 +189,15 @@ expr State::strip_undef_and_add_ub(const Value &val, const expr &e) {
 
   // two variants
   // 1) boolean
-  if (is_if_undef(e, a, b)) {
+  if (is_input(e, a, b)) {
     addUB(move(b));
     return a;
   }
 
-  auto has_undef = [&](const expr &e) {
```

Comment:
All uses of has_undef are negating the result, so I changed this to well_defined.